private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -345,35 +344,29 @@ func (p *moduleDataStore) deleteInvalidModuleData(
 			}
 		}()
 	}
-	moduleDir, err := os.Open(dirPath)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		// If the moduleDir was already closed, that is fine
-		if err := moduleDir.Close(); err != nil && err != fs.ErrClosed {
-			p.logger.Debug("failed_delete_close_module_dir", zap.Error(err))
-			retErr = multierr.Append(retErr, err)
+	var deleted map[string]struct{}
```

Comment:
It would be good to add a comment here that we're going to delete everything but the lock file (while the lock is held).